codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -475,7 +483,31 @@ let rec pruneAndMergeNextRegistered = (register: register, ~isMerged=false) => {
   | DynamicContractRegister({nextRegister})
     if register.latestFetchedBlock.blockNumber <
     nextRegister.latestFetchedBlock.blockNumber => merged
-  | DynamicContractRegister(_) =>
+  | DynamicContractRegister({nextRegister}) =>
+    if register.latestFetchedBlock.blockNumber > nextRegister.latestFetchedBlock.blockNumber {
```

Comment:
I want to validate whether this case is happening. It should never but I have know way of validating without adding a log. I could just add a log and stop it from crashing but it would be invalid if we ever hit this.

Registers should only merge when the latestFetchedBlockNumber is equal.